Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender.py

```diff
@@ -530,12 +530,15 @@ def main() -> None:
     # if your Client class inherits from BaseClient, system proxy is handled
     # out of the box by it, just pass ``proxy`` to the Client constructor
     proxy = params.get('proxy', False)
-    app_id = params.get('app_id')
+    app_id = params.get('app_id') or params.get('_app_id')
     base_url = params.get('base_url')
 
-    tenant_id = params.get('tenant_id')
+    tenant_id = params.get('tenant_id') or params.get('_tenant_id')
     client_credentials = params.get('client_credentials', False)
-    enc_key = params.get('enc_key')
+    enc_key = params.get('enc_key') or (demisto.getParam('credentials') or {}).get('password')
+
+    if not app_id:
+        raise Exception('Aplication ID must be provided.')
```

Comment:
what about tenant_id and enc_key?

Reply:
btw, let's put all of this inside the try block

Reply:
The tenant_id and enc_key are not required if the user used the device code authentication flow.